js/queue/queue-data-structure.js

```diff
@@ -5,22 +5,48 @@ class Queue {
   }
 
   canEnqueue() {
+
+    if (this.queueControl.length >= this.MAX_SIZE) {
+      return false
+    } else {
+      return true
+    }
     // ... your code goes here
   }
 
   isEmpty() {
+    if (this.queueControl.length === 0) {
+      return true
+    } else {
+      return false
+    }
     // ... your code goes here
   }
 
   enqueue(item) {
+    if (!this.canEnqueue()) {
+      throw new Error('QUEUE_OVERFLOW') 
+    }
+
+    this.queueControl.push(item)
+      
+      return this.queueControl
     // ... your code goes here
   }
 
   dequeue() {
+    if (!!this.isEmpty()) {
```

Comment:
El unico uso del !! es convertir un falsie a un false. Aquí es equivalente a `this.isEmpty()`